src/routes/api/product.routes.js

```diff
@@ -59,4 +62,27 @@ productRoutes.get(
   checkIfSellerOwnsProduct,
   getSpecificProduct
 );
+productRoutes.patch('/:productId',
+  protectRoute,
+  restrictTo('seller', 'admin'),
+  productValidation,
+  checkIfProductExistsById,
+  updateProduct
+);
+
+productRoutes.put(
+  '/:productId',
+  protectRoute,
+  restrictTo('seller','admin'),
+  checkIfProductIsAvailableById,
+  markAvailableProduct,
+);
+
+   productRoutes.delete(
+    '/:productId',
+    protectRoute,
+    restrictTo('seller', 'admin'),
+    checkIfProductExistsById,
+    deleteProduct
```

Comment:
You're missing a middleware to check if the products belong to the seller, and it's already implemented